src/tests/ftest/deployment/basic_checkout.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2018-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+from ior_small import IorSmall
+from mdtest_small import MdtestSmall
+from autotest import ContainerAutotestTest
+from ior_smoke import EcodIor
+from large_file import DmvrPosixLargeFile
+
+class BasicCheckout(IorSmall, MdtestSmall, ContainerAutotestTest, EcodIor):
```

Comment:
The problem with importing classes like `ContainerAutotestTest` is you are also adding test methods to this class, like https://github.com/daos-stack/daos/blob/master/src/tests/ftest/container/autotest.py#L18-L34 which mean that this tags for that test method - in this case `all,full_regression,daily_regression,hw,medium,ib2,container,autotest,containerautotest,quick` will result in the `deployment/basic_checkout.py:BasicCheckout.test_container_autotest` being run in daily runs on HW Medium but with your CI-incompatible deployment/basic_checkout.yaml file.

To get around this issue, you want a TestBase class setup like this:

- use the contents of the ContainerAutotestTest.test_container_autotest to define a method to run autotest indepently of the Test class in a new `ftest/util/autotest_test_base.py` file:

```
from apricot import TestWithServers
from command_utils_base import CommandFailure


class AutotestTestBase(pool, daos_cmd):
    # pylint: disable=too-few-public-methods
    """Tests container autotest.

    :avocado: recursive
    """

    def run_autotest(self):
        """Run container autotest."""
        self.log.info("Create a pool")
        self.add_pool()
        daos_cmd = self.get_daos_command()
        self.log.info("Autotest start")
        try:
            daos_cmd.pool_autotest(pool=self.pool.uuid)
            self.log.info("daos pool autotest passed.")
        except CommandFailure as error:
            self.log.error("Error: %s", error)
            self.fail("daos pool autotest failed!")

```

- modify container/autotest.py to inherit this test base class:

```
from autotest_test_base import AutotestTestBase


class ContainerAutotestTest(AutotestBaseClass):
    # pylint: disable=too-few-public-methods
    """Tests container autotest.

    :avocado: recursive
    """

    def test_container_autotest(self):
        """Test container autotest.

        :avocado: tags=all,full_regression,daily_regression
        :avocado: tags=hw,medium,ib2
        :avocado: tags=container,autotest,containerautotest,quick
        """
        self.run_autotest()

```

- Now the deployment/basic_checkout.py file can inherit the new `AutotestBaseClass` class instead of the `ContainerAutotestTest` class and use its `run_autotest()` method.